Assets/Scripts/Spawner/api/SpawnerChain.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+using System.Collections.Generic;
+using Creatures.Api;
+using Spawner.Api;
+using UnityEngine;
+
+namespace Spawner.Api
+{
+    public class SpawnerChain<T> where T : Creature
+    {
+        private GameObject spawnPoint;
+        private Stack<CreatureSpawner<T>> spawners;
+        private T creatureInstance;
+
+        public SpawnerChain(T creature, GameObject spawnPoint) {
+            this.spawners = new Stack<CreatureSpawner<T>>();
+            this.creatureInstance = Object.Instantiate(creature, spawnPoint.transform.position, Quaternion.identity);
+            this.spawnPoint = spawnPoint;            
+        }
+
+        public SpawnerChain<T> with(CreatureSpawner<T> spawner) {
+            spawners.Push(spawner); 
+            return this;
+        }
+
+        public T spawn()
+        {
+            while (spawners.Count > 0)
+            {
+                CreatureSpawner<T> spawner = spawners.Pop();
+                creatureInstance = spawner.apply(creatureInstance);
+            }
+            return creatureInstance;
+        }
+    }
+}
```

Comment:
is this newline missing an autoformat thing?

Reply:
fixt it